fairlearn/reductions/grid_search_classification.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import copy
+import numpy as np
+
+from fairlearn.metrics.demographic_parity import DemographicParity
+
+
+class GridSearchClassification:
+    def __init__(self,
+                 learner,
+                 fairness_metric=DemographicParity(),
+                 number_of_lagrange_multipliers=7):
+        self.learner = learner
+        if not isinstance(fairness_metric, DemographicParity):
+            raise RuntimeError("DemographicParity is only currently supported fairness metric")
+        self.fairness_metric = fairness_metric
+        self.number_of_lagrange_multipliers = number_of_lagrange_multipliers
+
+    def fit(self, X, Y, protected_attribute, lagrange_multipliers=None):
```

Comment:
Since the Lagrange multipliers can depend on the data, I thought it appropriate to put them here. Opinions?

Reply:
i think that this should be just fit(self, X, Y, **kwargs). then **kwargs should be passed to the fairness metrics and the objective.

Reply:
Done